ls1intum/Artemis
# [Feature] Grading System - Student Exam Summary

### Checklist

### Motivation and Context


This PR integrates the grading system in the student exam summary page. Students are now able to see the grade they've achieved on the exam (respectively the bonus points they've gained if the exam grades are of type bonus). The bonus points information is also now hidden, as long as there are no exercises that give bonus points (which is almost always the case). 

### Description

I've added some conditions to already present columns in the exam points summary component.
Also, I've added an additional section that displays the grade/bonus points of the student.
In a follow-up, a link will be created to the grade key, so a student can see the scale and where he/she fits into it.

### Steps for Testing


1. Log in to Artemis as an instructor
2. Set a grade key of type 'Grade' for an exam with student submissions in it.
3. Log in as one of the students with submissions in the said exam and go to the exam summary page (Course Overview -> Course -> Exams -> Exam name)
a) check if the corresponding grade is present (red if the student has failed the exam and green if he/she passed)
b) if none of the exam's exercises have bonus points, no column for bonus points should be displayed in the summary
c) test exporting (Colors don't get maintained unless the 'Background graphics' option is ticked in the export GUI).
d) check if any errors are getting displayed in the console/network tab
4. Now switch the grade ket to type 'Bonus' (again as an instructor)
5. Go to the same exam summary page and check the same steps in 3 a) - d), where instead of grade outcome a bonus points outcome should be displayed (without any colors).
6. Finally delete the grade key and go to the same overview page to test if any errors pop up if no grade key is present.

### Test Coverage



GradeStepResource.java: 100%
exam-points-summary.component.ts: 94.52%

### Screenshots


![Screenshot 2021-05-24 175701](https://user-images.githubusercontent.com/63168603/119404381-618d0b00-bce8-11eb-9221-f34bdc40c90b.png)
![Screenshot 2021-05-24 231345](https://user-images.githubusercontent.com/63168603/119404382-6225a180-bce8-11eb-8c45-9caf923b2a70.png)
![Screenshot 2021-05-24 175614](https://user-images.githubusercontent.com/63168603/119404383-62be3800-bce8-11eb-9ace-f3b42822961b.png)
